File: app/src/main/java/com/centrefx/idcardinfo/WarningDialog.java
package com.centrefx.idcardinfo;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class WarningDialog extends DialogFragment
{
    private final String message;

    public WarningDialog(String msg)
    {
        this.message = msg;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
    {
        AlertDialog.Builder ad = new AlertDialog.Builder(requireContext());
        ad.setTitle("Error!");
        ad.setCancelable(true);
        ad.setMessage(message);
        ad.setPositiveButton("OK", (dialog, which) -> { });
        return ad.create();
    }
}

File: app/src/main/java/com/centrefx/idcardinfo_k/WarningDialog.kt
package com.centrefx.idcardinfo_k

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class WarningDialog(private val msg: String) : DialogFragment() {

    override fun onCreateDialog(
        savedInstanceState: Bundle?
    ): Dialog {
        val ad: AlertDialog.Builder = AlertDialog.Builder(requireContext())
        ad.setTitle("Error!")
        ad.setCancelable(true)
        ad.setMessage(msg)
        ad.setPositiveButton("") { _: DialogInterface, _: Int -> }
        return ad.create()
    }
}